src/services/Elements.php

```diff
@@ -3241,17 +3248,71 @@ private function _propagateElement(
 
         // Save it
         $siteElement->setScenario(Element::SCENARIO_ESSENTIALS);
+
+        // for Entries and Categories: validate element against "live" scenario across all sites,
+        // if element is to be set to live/pending/enabled
+        if ($crossSiteValidate && ($siteElement instanceof Entry || $siteElement instanceof Category)) {
+            $siteElementStatus = $siteElement->getStatus();
+            if (($siteElement instanceof Entry &&
+                    ($siteElementStatus === Entry::STATUS_LIVE || $siteElementStatus === Entry::STATUS_PENDING)
+                ) ||
+                ($siteElement instanceof Category && $siteElementStatus === Category::STATUS_ENABLED)
+            ) {
+                $siteElement->setScenario(Element::SCENARIO_LIVE);
+            }
```

Comment:
We should be respecting the original element’s validation scenario, regardless of status.

Reply:
This was added to mitigate issue: #11569, so the whole point was to validate againt a Live scenario, if the Element is set to be live or pending or enabled (in case of e.g. categories), right? Or am I missing something here?

Without this, it's validated against only essentials, unless someone changes the scenario via e.g. `Entry::EVENT_BEFORE_SAVE` (like in the example from that issue).

Reply:
Since this is happening in a private method (`_propagateElement()`) and we have control over all of the places it’s called, we should remove this line entirely from it:

```php
$siteElement->setScenario(Element::SCENARIO_ESSENTIALS);
```

And start calling `setScenario()` on the passed-in `$siteElement` in each of the places that `_propagateElement()` is called.

Generally that will still be `Element::SCENARIO_ESSENTIALS`, except for `_saveElementInternal()`. In that latter case, **_if_** `$crossSiteValidate` is `true`, set it to the same scenario as `$element` instead:

```php
if ($crossSiteValidate) {
    $siteElement->setScenario($element->getScenario());
} else {
    $siteElement->setScenario(Element::SCENARIO_ESSENTIALS);
}
```

Reply:
I don't think we can do `$siteElement->setScenario($element->getScenario());` because what if you have 3 sites (propagation set to 'all'), you start creating an entry on Site 1, you trigger cross site validation and it returns an error for e.g. Site 2; you go to that entry for Site 2 and disable it - in that case `$siteElement->setScenario($element->getScenario());` will still insist on validating against live scenario even though you want `$siteElement` to be disabled. I still think it needs to check the status of the `$siteElement` and only use live scenario if that element is set to be live, pending or enabled.

I could move scenario setting to each place we call `_propagateElement()`, but `_propagateElement()` is the only place where we can be sure `$siteElement` is actually set, so having `Element::SCENARIO_ESSENTIALS` in there is a nice safety net - or am I overthinking this part?